sources/engine/Stride.Engine/Profiling/GameProfilingSystem.cs

```diff
@@ -323,9 +322,9 @@ protected override void Destroy()
 
             Profiler.Unsubscribe(profilerChannel);
 
-            if (stringBuilderTask != null && !stringBuilderTask.IsCompleted)
+            if (asyncUpdateTask != null && !asyncUpdateTask.IsCompleted)
             {
-                stringBuilderTask.Wait();
+                asyncUpdateTask.Wait();
```

Comment:
Any long-running operation should have an associated `TaskCancellationSource` and pass the associated `CancellationToken` everywhere. `Destroy()` should not hang on the next `RefreshTime` tick.